src/sentry/api/event_search.py

```diff
@@ -847,7 +847,7 @@ def to_list(value):
     "sum": {"snuba_name": "sum", "fields": ["transaction.duration"]},
 }
 
-AGGREGATE_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<column>[a-z\._]*)\)$")
+AGGREGATE_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<column>[a-z\._-]*)\)$")
```

Comment:
@dashed columns can have numbers

Reply:
Tag keys can have any character

Reply:
We should update the patterns used here to accept the full character range. Does the pattern in the client side code need to be updated as well?

Reply:
I'll check the client side.

Reply:
Snuba has a schema validation step on the tag names: https://github.com/getsentry/snuba/blob/5e5de9e85b68fb931c7e9711db5c873774363dcf/snuba/query/schema.py#L117-L127

I'm going to update based on this.